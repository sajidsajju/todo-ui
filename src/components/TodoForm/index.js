import React, { useState } from "react";
import { makeStyles, TextField } from "@material-ui/core";
import firebase from "../../util/firebase";
import { addTodo } from "../Api";

const useStyles = makeStyles({
  form: {
    textAlign: "center",
    width: "40em",
    margin: "auto",
  },
  textField: {
    "&&&:before": {
      borderBottom: "none",
    },
    "&&:after": {
      borderBottom: "none",
    },
    color: "#000000",
    borderRadius: "25px",
    border: "5px solid #696969",
    fontSize: "1.4em",
    paddingLeft: "10px",
    background: "#FFFFFF",
    overflow: "hidden",
  },
});

function TodoForm(props) {
  const classes = useStyles();
  const { user } = props;
  const [text, setText] = useState("");

  const handleChange = event => {
    if (event.target.value.length < 30) {
      const value = event.target.value;
      setText(value);
    }
  };

  const onSubmit = event => {
    event.preventDefault();
    const todo = {
      date: firebase.firestore.FieldValue.serverTimestamp(),
      text: text,
      completed: false,
    };
    addTodo(user.uid, todo);
    setText("");
  };

  return (
    <form className={classes.form} onSubmit={onSubmit}>
      <TextField
        InputProps={{ className: classes.textField }}
        margin="normal"
        placeholder="what needs to be added ? "
        fullWidth
        value={text}
        onChange={handleChange}
        autoComplete="off"
        required
        data-testid="input-field"
      />
    </form>
  );
}

export default TodoForm;
