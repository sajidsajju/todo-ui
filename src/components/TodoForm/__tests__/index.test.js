import React from "react";
import { fireEvent, render } from "@testing-library/react";

import TodoForm from "../";

describe("<TodoForm  />", () => {
  const user = { uid: "12" };

  it("should match the snapshot", () => {
    const rendered = render(<TodoForm user={user} />);

    expect(rendered.container.firstChild).toMatchSnapshot();
  });

  it("should check the empty text field", () => {
    const { getByTestId } = render(<TodoForm user={user} />);

    const input = getByTestId("input-field").querySelector("input");

    expect(input).toHaveValue("");
  });

  it("should check the text field with some text", () => {
    const { getByTestId } = render(<TodoForm user={user} />);

    const input = getByTestId("input-field").querySelector("input");

    fireEvent.change(input, { target: { value: "hello" } });

    expect(input).toHaveValue("hello");
  });
});
