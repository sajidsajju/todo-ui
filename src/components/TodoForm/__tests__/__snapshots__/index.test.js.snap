// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoForm  /> should match the snapshot 1`] = `
<form
  class="makeStyles-form-1"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    data-testid="input-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
    >
       
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-textField-2 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        autocomplete="off"
        class="MuiInputBase-input MuiInput-input"
        placeholder="what needs to be added ? "
        type="text"
        value=""
      />
    </div>
  </div>
</form>
`;
